docs/design/2024-08-07-auto-pass-through-hashagg.md

```diff
@@ -0,0 +1,54 @@
+# AutoPassThroughHashAgg Design Doc
+
+- Author(s): [guo-shaoge](http://github.com/guo-shaoge)
+- Tracking Issue: https://github.com/pingcap/tiflash/issues/9196
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Detailed Design](#detailed-design)
+* [Impacts & Risks](#impacts--risks)
+* [Unresolved Questions](#unresolved-questions)
+
+## Introduction
+The HashAgg pushed down to TiFlash can be a one-stage, two-stage, or three-stage. For two-stage and three-stage aggregations, the 1st hashagg is used for pre-aggregation to reduce the amount of data that needs to be shuffled.
```

Comment:
```suggestion
The HashAgg pushed down to TiFlash can be a plan of one-stage, two-stage, or three-stage. For two-stage and three-stage aggregations, the 1st hashagg is used for pre-aggregation to reduce the amount of data that needs to be shuffled.
```

Reply:
I guess you are using "hashagg" (all lower-cased) to differentiate the plan level term "HashAgg" (camel-cased). If I'm guessing right, you may just use "aggregation" because "hashagg" is not a valid word.

Reply:
done. All `hashagg` changed to `aggregation`